### ISSUE TITLE
global top variables of a module not exported

### ISSUE THREAD

**Issue by pavelkveton** [qorelanguage/qore#1673]
the documentation https://qoretechnologies.com/manual/qorus/latest/qore/lang/html/qore_modules.html says:

> Any namespaces, classes, constants, functions, and global variables declared as public will be exported into the importing Program object; all other declarations will be private to the module's Program object. Note that the root namespace of a module Program is always public by default.

Now consider the following module
```
%requires qore >= 0.8.12                                                                 
                                                                                         
%new-style
                                                                                         
module test {                                                                            
    version = "0.3.11.1";                                                                
    desc = "HttpServer class definition";                                                
    author = "David Nichols <david@qore.org>";
    url = "http://qore.org";
    license = "MIT";
}

our string zzz;
```
and code:
```
%new-style

%requires ./test.qm

zzz = "aaa";
printf("%y\n", zzz);
```
running it, it produces the error:
```
unhandled QORE System exception thrown in TID 1 at 2017-02-16 09:25:06.636640 Thu +01:00 (CET) at test.q:5
PARSE-EXCEPTION: cannot resolve bareword 'zzz' to any reachable object
```
When you change `our string zzz` to `public our string zzz`, it start working. I think it is a bug somewhere - are top-level global variables in a module really considered public as the documentation says?

**Comment by davidnich** [comment]
the documentation says that the root namespace is `public`, not that definitions in that namespace are automatically `public` as well.

If you have a `public` namespace and do not declare the definitions in the namespace also as `public`, then those definitions not declared `public` will not be exported.  In order to have a definition exported, the enclosing namespace has to be `public` and also the definition itself.  

If you declare your variable definition above as `public` then it should work.

In other words the documentation doesn't say what you assumed, but we can use this issue to add a clarification in any case as it's obviously not totally clear.

**Comment by omusil24** [comment]
This is pretty unclear IMO. Since you can't change the publicness/privateness of the root namespace and it's always public, which is how it intuitively should be, then I would remove that information to not confuse people. That first sentence should be enough.
